### PR TITLE
Book list ✨

### PR DESCRIPTION
**Pull request by moralesbang**
### Summary

Added book listing using book card component.

### Screenshots

Screen Shot 2020-05-18 at 4 33 40 PM

### Trello Card

https://trello.com/c/FPPQwCf1

### CODE REVIEW

File: src/app/screens/BookList/index.tsx [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 import BookCard from '@components/BookCard';
+import { BOOKS } from '@constants/mockData';
+import { Book } from '@interfaces/book';
 
 import styles from './styles';
 
+const renderBook: ListRenderItem<Book> = ({ item: { title, author, imageURL } }) => (
+  <BookCard title={title} author={author} coverImageURI={imageURL} />
+);
+
```
Comment (sfernandez11) [inline src/app/screens/BookList/index.tsx:12]: Why outside the function declaration?

File: src/app/screens/BookList/index.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
-      <BookCard
-        title="Scavenger Hunt"
-        author="Christopher Pike"
-        coverImageURI="http://wolox-training.s3.amazonaws.com/uploads/6963511-M.jpg"
-      />
-    </View>
+    <FlatList
+      data={BOOKS}
+      renderItem={renderBook}
+      keyExtractor={({ id }) => String(id)}
```
Comment (sfernandez11) [inline src/app/screens/BookList/index.tsx:18]: Extract this function => ```const keyExtractor = ({ id }: Book) => `${id}`;```

File: src/app/screens/BookList/index.tsx [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
 function BookList() {
   return (
-    <View style={styles.bookList}>
-      <BookCard
-        title="Scavenger Hunt"
-        author="Christopher Pike"
-        coverImageURI="http://wolox-training.s3.amazonaws.com/uploads/6963511-M.jpg"
-      />
-    </View>
+    <FlatList
```
Comment (sfernandez11) [inline src/app/screens/BookList/index.tsx:15]: `<FlatList<Book>`

File: src/constants/mockData.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Book } from '@interfaces/book';
+
+export const BOOKS: Book[] = [
```
Comment (sfernandez11) [inline src/constants/mockData.ts:3]: Why are books missing from this list? Check this [JSON](https://docs.google.com/document/d/1vzK9M8Dhh5hdRNQPWpuCa1-vF-EO3sGO971-jm3Kli0)

File: src/app/screens/BookList/index.tsx (hunk 1 above)
Comment (moralesbang) [reply src/app/screens/BookList/index.tsx:12]: @marianozicavo

Comment (moralesbang) [reply src/app/screens/BookList/index.tsx:12]: I don't have reason. I'm gonna move it.

File: src/constants/mockData.ts (hunk 4 above)
Comment (moralesbang) [reply src/constants/mockData.ts:3]: Fixe it.

File: src/app/components/BookCard/index.tsx [hunk 5]
```diff
@@ -1,18 +1,20 @@
...
+  bookData: Book;
 }
 
-function BookCard({ title, author, coverImageURI }: Props) {
+function BookCard({ bookData: { title, author, imageUrl } }: Props) {
   return (
     <View style={styles.book}>
-      <Image source={{ uri: coverImageURI }} style={styles.bookCover} />
+      <Image
+        source={imageUrl ? { uri: imageUrl } : require('@assets/book_placeholder_cover.png')}
```
Comment (marianozicavo) [inline src/app/components/BookCard/index.tsx:15]: Import this like the other variables.
like `import icCheck from '@assets/icCheck.png';`

File: src/app/components/BookCard/index.tsx [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
 
 interface Props {
-  title: string;
-  author: string;
-  coverImageURI: string;
+  bookData: Book;
 }
 
-function BookCard({ title, author, coverImageURI }: Props) {
+function BookCard({ bookData: { title, author, imageUrl } }: Props) {
```
Comment (sfernandez11) [inline src/app/components/BookCard/index.tsx:12]: And here => `function BookCard({ title, author, imageUrl }: Props) {`

File: src/app/screens/BookList/index.tsx [hunk 7]
```diff
@@ -1,18 +1,22 @@
...
-import { View } from 'react-native';
+import { FlatList, ListRenderItem } from 'react-native';
 import BookCard from '@components/BookCard';
+import { BOOKS } from '@constants/mockData';
+import { Book } from '@interfaces/book';
 
 import styles from './styles';
 
+export const keyExtractor = ({ id }: Book) => String(id);
+export const renderBook: ListRenderItem<Book> = ({ item }) => <BookCard bookData={item} />;
```
Comment (sfernandez11) [inline src/app/screens/BookList/index.tsx:10]: Sorry but this is still wrong, please check the previous comments.
Why is this still outside the function declaration?
Also => `const renderBook: ListRenderItem<Book> = ({ item }) => <BookCard {...item} />;`

File: src/app/components/BookCard/index.tsx [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
+import bookPlaceholderCover from '@assets/book_placeholder_cover.png';
 
 import styles from './styles';
 
 interface Props {
-  title: string;
-  author: string;
-  coverImageURI: string;
+  bookData: Book;
 }
```
Comment (sfernandez11) [inline src/app/components/BookCard/index.tsx:10]: `interface Props extends Book {}` for now

Comment (moralesbang) [reply src/app/components/BookCard/index.tsx:10]: Cool. I didn't know that.

File: src/app/screens/BookList/index.tsx (hunk 7 above)
Comment (moralesbang) [reply src/app/screens/BookList/index.tsx:10]: Yeat. I missed up.

**Pull request merged**